tools/cli/data_converter_commands.go

```diff
@@ -0,0 +1,137 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/urfave/cli"
+
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/server/tools/cli/dataconverter"
+)
+
+type PayloadRequest struct {
+	RequestID string `json:"requestId"`
+	Payload   string `json:"payload"`
+}
+
+type PayloadResponse struct {
+	RequestID string `json:"requestId"`
+	Content   string `json:"content"`
+}
+
+func processMessage(c *websocket.Conn) error {
+	mt, message, err := c.ReadMessage()
+	if err != nil {
+		if closeError, ok := err.(*websocket.CloseError); ok {
+			if closeError.Code == websocket.CloseNoStatusReceived ||
+				closeError.Code == websocket.CloseNormalClosure {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("error reading websocket message: %w", err)
+	}
+
+	var payloadRequest PayloadRequest
+	err = json.Unmarshal(message, &payloadRequest)
+	if err != nil {
+		return fmt.Errorf("invalid payload request: %w", err)
+	}
+
+	var payload commonpb.Payload
+	err = jsonpb.UnmarshalString(payloadRequest.Payload, &payload)
+	if err != nil {
+		return fmt.Errorf("invalid payload data: %w", err)
+	}
+
+	payloadResponse := PayloadResponse{
+		RequestID: payloadRequest.RequestID,
+		Content:   dataconverter.GetCurrent().ToString(&payload),
+	}
+
+	var response []byte
+	response, err = json.Marshal(payloadResponse)
+	if err != nil {
+		return fmt.Errorf("unable to marshal response: %w", err)
+	}
+
+	err = c.WriteMessage(mt, response)
+	if err != nil {
+		return fmt.Errorf("unable to write response: %w", err)
+	}
+
+	return nil
+}
+
+func buildPayloadHandler(context *cli.Context, origin string) func(http.ResponseWriter, *http.Request) {
+	upgrader := websocket.Upgrader{
+		CheckOrigin: func(r *http.Request) bool {
+			if r.Header.Get("Origin") != origin {
+				fmt.Printf("invalid origin: %s\n", origin)
+				return false
+			}
+			return true
+		},
+	}
+
+	return func(res http.ResponseWriter, req *http.Request) {
+		c, err := upgrader.Upgrade(res, req, nil)
+		if err != nil {
+			fmt.Printf("websocket upgrade failed: %v\n", err)
+			return
+		}
+		defer c.Close()
+
+		for {
+			if err := processMessage(c); err != nil {
+				return
+			}
+		}
+	}
+}
+
+// DataConverter provides a data converter over a websocket for Temporal web
+func DataConverter(c *cli.Context) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		ErrorAndExit("Unable to create listener", err)
+	}
+	origin := c.String(FlagWebURL)
+	port := listener.Addr().(*net.TCPAddr).Port
+	url := origin + "/data-converter/" + fmt.Sprint(port)
```

Comment:
Can you use `fmt.Sprintf` here and extract `"%s/data-converter/%d"` as a `const` in `dtaconverter` package. It seems to be a part of a contract.